rs/crypto/internal/crypto_service_provider/src/vault/local_csp_vault/multi_sig/tests.rs

```diff
@@ -207,49 +208,106 @@ fn should_multi_sign_and_verify_with_generated_key() {
         .is_ok());
 }
 
+#[test]
+fn should_fail_to_sign_if_committee_signing_public_key_not_found() {
+    let mut pks_returning_none = MockPublicKeyStore::new();
+    pks_returning_none
+        .expect_committee_signing_pubkey()
+        .return_const(None);
+    let vault = LocalCspVault::builder_for_test()
+        .with_public_key_store(pks_returning_none)
+        .build();
+
+    let result = vault.multi_sign(AlgorithmId::MultiBls12_381, b"message".to_vec());
+
+    assert_eq!(result, Err(CspMultiSignatureError::PublicKeyNotFound));
+}
+
+#[test]
+fn should_fail_to_sign_if_key_id_instantiation_fails() {
+    // This would only fail if the committee signing public key is larger than 4 GB.
+    // Given that creating such large keys in a test is very expensive, and
+    // that this scenario is extremely unlikely (as the node key generation
+    // logic has to be broken) we are not explicitly testing this.
```

Comment:
I had been wondering (having forgotten) why the PK -> KeyId conversion fails.

I think we could make this a bit nicer by defining KeyId also for large keys:

* If the key length fits into 32 bits, perform exactly as we do now
* If the key length does not fit into 32 bits, do something else (basically just use a 64-bit length plus an alternate domain separator)

Then `KeyId::from(pk)` is infallible and quite a few error cases go away. Created CRP-2687 for this